azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/transport/mqtt/mqtt_transport.py

```diff
@@ -10,6 +10,7 @@
 from .mqtt_provider import MQTTProvider
 from transitions.extensions import LockedMachine as Machine
 from azure.iot.hub.devicesdk.transport.abstract_transport import AbstractTransport
+from azure.iot.hub.devicesdk.transport import constant
```

Comment:
>constant [](start = 46, length = 8)

cant we declare constants here ? 
or else it is better tot take all constants from here to the constant module.

Reply:
These constants apply to the entire transport package, so I figured they ought to be defined there, as they are used elsewhere as well.

When used elsewhere you import them the same way - from azure.iot.hub.devicesdk.transport import constant, then you refer to the constants as, say constant.C2D (see: tests, message_queue, etc.)

I find this syntax preferable to exposing them through the init by doing "from azure.iot.hub.devicesdk.transport import C2D" as the constant module adds clarity.

I'm open to alternatives if anyone has strong feelings about this though.

---
In reply to: [259174246](https://github.com/Azure/azure-iot-sdk-python-preview/pull/49#discussion_r259174246) [](ancestors = 259174246)

Reply:
I like having all constants in the same place. easier to debug/find them later. whether these should be in an enum or "floating" constants I'm not sure. pick the most pythonic way?

---
In reply to: [259175173](https://github.com/Azure/azure-iot-sdk-python-preview/pull/49#discussion_r259175173) [](ancestors = 259175173,259174246)

Reply:
Enum isn't really a thing in standard Python prior to 3.4. Python originally just used string literals for this purpose. For compat reasons I think that's the route we should continue, rather than depending on an external enum library.

---
In reply to: [259537743](https://github.com/Azure/azure-iot-sdk-python-preview/pull/49#discussion_r259537743) [](ancestors = 259537743,259175173,259174246)